packages/xo-web/src/xo-app/settings/cloud-configs/index.js

```diff
@@ -87,90 +106,167 @@ export default decorate([
           }
           reset()
         },
+      editNetworkCloudConfig:
+        ({ reset }) =>
+        async ({ networkName, networkTemplate, networkCloudConfigToEditId }, { cloudConfigs }) => {
+          const oldCloudConfig = find(cloudConfigs, { id: networkCloudConfigToEditId })
+          if (oldCloudConfig.name !== networkName || oldCloudConfig.template !== networkTemplate) {
+            await editNetworkCloudConfig(networkCloudConfigToEditId, { name: networkName, template: networkTemplate })
+          }
+          reset()
+        },
       populateForm:
-        (_, { id, name, template }) =>
+        (_, { id, name, template, type }) =>
         state => ({
           ...state,
-          name,
-          cloudConfigToEditId: id,
-          template,
+          [type === 'network' ? 'networkName' : 'name']: name,
+          [type === 'network' ? 'networkCloudConfigToEditId' : 'cloudConfigToEditId']: id,
+          [type === 'network' ? 'networkTemplate' : 'template']: template,
         }),
     },
     computed: {
       formId: generateId,
       inputNameId: generateId,
       inputTemplateId: generateId,
       isInvalid: ({ name, template }) => name.trim() === '' || (template !== undefined && template.trim() === ''),
+      isNetworkValid: ({ networkName, networkTemplate }) =>
+        networkName.trim() === '' || (networkTemplate !== undefined && networkTemplate.trim() === ''),
```

Comment:
I think its isNetwork_IN_valid . If not , I think you should add a comment on why the divergence with isINvalid